### DISCUSSION TITLE
Installation problem on MacOS Ventura 13.5 (i7 processor)

### DISCUSSION THREAD

**Discussion by grarddeg** [Audiveris/audiveris#661] [Q&A] [answered]
Hi,

I followed the instructions on https://github.com/Audiveris/audiveris/wiki/Installation
First cloned, then installed JDK (latest version 20.0.2 as seen in the output of java -version below
java version "20.0.2" 2023-07-18
Java(TM) SE Runtime Environment (build 20.0.2+9-78)
Java HotSpot(TM) 64-Bit Server VM (build 20.0.2+9-78, mixed mode, sharing)

Build returns an error message:
Could not compile settings file '/Users/grarddeg/Applications/audiveris/settings.gradle'.
> startup failed:
  General error during conversion: Unsupported class file major version 64

Any suggestion on how to solve the problem ?

**Comment by grarddeg** [comment, marked as answer]
Problem solved by downgrading to JDK 17 (build 17.0.8+9-LTS-211)

**Comment by hbitteur** [comment]
@grarddeg 
Your problem sounds like a Java version mismatch.
How did you attempt to build? (with Java 20 and with Java 17)

For your information, the file `build.gradle` contains the line:
> sourceCompatibility = '1.17'

Did you try to modify this line?

I have no access to a Mac, and thus can't help you concretely.

**Comment by grarddeg** [comment]
I first tried to build with java 20, and that failed.
I then uninstalled java 20 and installed java 17 instead.
I tried to build again with java 17 and that worked :-)

**Comment by hbitteur** [comment]
Yes, current Audiveris version is meant to work with the latest Java LTS version, which means Java 17 today.
And your build worked as expected. Good point!

Now, I would like to understand the initial problem.
I don't know with Java version corresponds to "class file major version 64".

My understanding is that you did not try to modify the `build.gradle` file from 
- sourceCompatibility = '1.17'
to
- sourceCompatibility = '1.20'

Do you confirm?

**Comment by hbitteur** [comment]
This just crossed my mind:
Perhaps Gradle also needs to be upgraded for Java 20?

See [https://stackoverflow.com/questions/75878942/spring-initializr-project-results-in-unsupported-class-file-major-version-64](https://stackoverflow.com/questions/75878942/spring-initializr-project-results-in-unsupported-class-file-major-version-64)

**Reply by grarddeg** [reply]
I didn't modify build.gradle indeed, so I wouldn't be able to tell you whether gradle is compatible with Java 20.
